components/script/textinput.rs

```diff
@@ -850,7 +857,7 @@ impl<T: ClipboardProvider> TextInput<T> {
         } else {
             vec!(content)
         };
-        if update_text_cursor {
+        if update_text_cursor == ChangeEditPoint::Change {
```

Comment:
I'm wondering if we can use `if let` here? Then maybe we won't need to `clone()` for the `update_text_cursor` argument in `htmltextareaelement.rs`.

```rust
pub fn set_content(&mut self, content: DOMString, update_text_cursor: &ChangeEditPoint) {
    ....
    if let ChangeEditPoint::Change = *update_text_cursor {
        self.edit_point.line = min(self.edit_point.line, self.lines.len() - 1);
        self.edit_point.index = min(self.edit_point.index, self.current_line_length());
   }
    ....
}
```

If the way in above example is fine, I think it would be great to also use the same way for `clear_selection_to_limit`! :)